drivers/scheduler/k8s/k8s.go

```diff
@@ -2709,6 +2742,13 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 			if err != nil {
 				return nil, err
 			}
+			related, err := k.relatedVolume(pvcObj)
+			if err != nil {
+				return nil, err
+			}
+			if !related {
+				continue
+			}
```

Comment:
Not sure if all volumes will have a backend. This will get called for non-pure volumes as well. In that case it would not return that volume.

Reply:
For cases of Torpedo running on non-Pure backends, the "relatedVolume" check will do a fast exit. If we have Pure backends enabled, I _think_ we want to always only include Pure volumes, at least in terms of the current functionality I'm refactoring.

Reply:
Got it. I did not realize that we are using`k.PureVolumes` to just run pure volume tests.